vllm/lora/ops/ipex_ops/lora_ops.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+import torch
+
+from vllm.logger import init_logger
+from vllm.platforms import current_platform
+from vllm.utils import direct_register_custom_op
+
+logger = init_logger(__name__)
+
+try:
+    import intel_extension_for_pytorch as ipex
+except ImportError as e:
+    logger.warning("Import error msg: %s", e.msg)
+
+
+def _bgmv_shrink(inputs: torch.Tensor,
+                 lora_a_weights: torch.Tensor,
+                 output_tensor: torch.Tensor,
+                 lora_indices_tensor: torch.Tensor,
+                 scaling: float = 1.0) -> None:
+
+    ipex.llm.functional.bgmv_shrink(inputs, lora_a_weights, output_tensor,
+                                    lora_indices_tensor, scaling)
+
+
+def _bgmv_shrink_fake(inputs: torch.Tensor,
+                      lora_a_weights: torch.Tensor,
+                      output_tensor: torch.Tensor,
+                      lora_indices_tensor: torch.Tensor,
+                      scaling: float = 1.0) -> None:
+    pass
+
+
+def _bgmv_expand(inputs: torch.Tensor,
+                 lora_b_weights: torch.Tensor,
+                 output_tensor: torch.Tensor,
+                 lora_indices_tensor: torch.Tensor,
+                 add_inputs: bool = True) -> None:
+    ipex.llm.functional.bgmv_expand(inputs, lora_b_weights, output_tensor,
+                                    lora_indices_tensor, add_inputs)
+
+
+def _bgmv_expand_fake(inputs: torch.Tensor,
+                      lora_b_weights: torch.Tensor,
+                      output_tensor: torch.Tensor,
+                      lora_indices_tensor: torch.Tensor,
+                      add_inputs: bool = True) -> None:
+    pass
+
+
+def _bgmv_expand_slice(inputs: torch.Tensor,
+                       lora_b_weights: torch.Tensor,
+                       output_tensor: torch.Tensor,
+                       lora_indices_tensor: torch.Tensor,
+                       slice_offset: int,
+                       slice_size: int,
+                       add_inputs: bool = True) -> None:
+    ipex.llm.functional.bgmv_expand_slice(inputs, lora_b_weights,
+                                          output_tensor, lora_indices_tensor,
+                                          slice_offset, slice_size, add_inputs)
+
+
+def _bgmv_expand_slice_fake(inputs: torch.Tensor,
+                            lora_b_weights: torch.Tensor,
+                            output_tensor: torch.Tensor,
+                            lora_indices_tensor: torch.Tensor,
+                            slice_offset: int,
+                            slice_size: int,
+                            add_inputs: bool = True) -> None:
+    pass
+
+
+try:
+    direct_register_custom_op(
```

Comment:
Do we need such wrapper as we actually not support compile on xpu for this feature? Can you just follow `punica_gpu.py` and adopt ipex kernel.

Reply:
reverted